private/pkg/app/feature_container.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"strconv"
+	"strings"
+)
+
+func newFeatureContainer(env EnvContainer) FeatureContainer {
```

Comment:
I think this dovetails with my above comment - this doesn't add anything above an `EnvContainer`. Everything at the top-level is stuff that maps to the OS cleanly - args, i/o, and enviornment variables. `FeatureContainer` is just an abstraction on top of an `EnvContainer`, it isn't a primitive (and doesn't feel necessary to do what you are doing here)

Reply:
Removed.